http3/http_stream.go

```diff
@@ -238,11 +238,7 @@ func (s *requestStream) ReadResponse() (*http.Response, error) {
 
 	// Check that the server doesn't send more data in DATA frames than indicated by the Content-Length header (if set).
 	// See section 4.1.2 of RFC 9114.
-	contentLength := int64(-1)
-	if _, ok := res.Header["Content-Length"]; ok && res.ContentLength >= 0 {
-		contentLength = res.ContentLength
-	}
-	respBody := newResponseBody(s.stream, contentLength, s.reqDone)
+	respBody := newResponseBody(s.stream, res.ContentLength, s.reqDone)
 
 	// Rules for when to set Content-Length are defined in https://tools.ietf.org/html/rfc7230#section-3.3.2.
 	_, hasTransferEncoding := res.Header["Transfer-Encoding"]
```

Comment:
Shouldn't we ignore this header?

Reply:
Why?

Reply:
https://github.com/quic-go/quic-go/issues/4622#issuecomment-2313198937

Reply:
@bt90 I think this you're correct. This was introduced by @WeidiDeng. @WeidiDeng, can you please check? I think the HTTP/3 RFC is pretty unambiguous with respect to the Transfer-Encoding. What was the motivation for adding this?

Reply:
![image](https://github.com/user-attachments/assets/3243bc77-2a3d-4a66-9fd4-fac2156e4858)

This change was not made by me. But I am pretty sure `Transfer-Encoding` is not necessary in http/2 & http/3 because of the data frame wire format. In caddy, we were discussing the meaning of the `Transfer-Encoding` for responses to range requests in the 1.23 [changelog](https://tip.golang.org/doc/go1.23). And I found out it's not possible for http2 & http3.

Reply:
I see, you just moved them in https://github.com/quic-go/quic-go/pull/4412. Sorry for the noise.

Reply:
Fix in #4646.